pkg/amqp/publisher.go

```diff
@@ -11,29 +12,62 @@ import (
 )
 
 type Publisher struct {
-	*connectionWrapper
+	*ConnectionWrapper
 
 	config                  Config
 	publishBindingsLock     sync.RWMutex
 	publishBindingsPrepared map[string]struct{}
+	closeConnection         func() error
 }
 
-func NewPublisher(config Config, logger watermill.LoggerAdapter) (*Publisher, error) {
+func NewPublisher(config Config, logger watermill.LoggerAdapter, opts ...Opt) (*Publisher, error) {
 	if err := config.ValidatePublisher(); err != nil {
 		return nil, err
 	}
 
-	conn, err := newConnection(config, logger)
-	if err != nil {
+	p := &Publisher{
+		config:                  config,
```

Comment:
we are avoiding named parameters with struct initialization to not forget to pass new fields. IMO it's better to pass zero values but do it explicitly.